web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/DatastreamController.java

```diff
@@ -136,6 +139,10 @@ public ResponseEntity<InputStreamResource> getThumbnail(@PathVariable("pid") Str
 
         PID pid = PIDs.get(pidString);
         AccessGroupSet principals = getAgentPrincipals().getPrincipals();
+        accessControlService.assertHasAccess("Insufficient permissions to get thumbnail for " + pidString,
+                pid, principals, Permission.viewAccessCopies);
+
+        size = size.toLowerCase();
         if (!size.equals(SMALL) && !size.equals(LARGE)) {
```

Comment:
just mentioning we could do this with `THUMB_SIZE_MAP.containsKey(size)`, but it's fine either way

Reply:
ohh thanks!